fe/fe-core/src/main/java/com/starrocks/load/streamload/StreamLoadMgr.java

```diff
@@ -212,27 +203,27 @@ public StreamLoadTask createLoadTaskWithoutLock(Database db, String tableName, S
         return streamLoadTask;
     }
 
-    public StreamLoadTask createLoadTask(Database db, String tableName, String label, String user, String clientIp,
+    public StreamLoadTask createLoadTask(Database db, Table table, String label, String user, String clientIp,
                                          long timeoutMillis, int channelNum,
-                                         int channelId, long warehouseId) throws StarRocksException {
-        Table table;
+                                         int channelId, long warehouseId) {
+        // init stream load task
+        long id = GlobalStateMgr.getCurrentState().getNextId();
+        StreamLoadTask streamLoadTask = new StreamLoadTask(id, db, (OlapTable) table,
+                label, user, clientIp, timeoutMillis, channelNum, channelId, System.currentTimeMillis(), warehouseId);
+        return streamLoadTask;
+    }
+
+    private Table checkMeta(Database db, String tableName) throws StarRocksException {
         Locker locker = new Locker();
         locker.lockDatabase(db.getId(), LockType.READ);
         try {
-            unprotectedCheckMeta(db, tableName);
-            table = GlobalStateMgr.getCurrentState().getLocalMetastore().getTable(db.getFullName(), tableName);
+            return unprotectedCheckMeta(db, tableName);
         } finally {
             locker.unLockDatabase(db.getId(), LockType.READ);
         }
-
-        // init stream load task
-        long id = GlobalStateMgr.getCurrentState().getNextId();
-        StreamLoadTask streamLoadTask = new StreamLoadTask(id, db, (OlapTable) table,
-                label, user, clientIp, timeoutMillis, channelNum, channelId, System.currentTimeMillis(), warehouseId);
-        return streamLoadTask;
     }
 
-    public void unprotectedCheckMeta(Database db, String tblName)
+    public Table unprotectedCheckMeta(Database db, String tblName)
```

Comment:
possibly change the interface to private, if the change of interface signature doesn't affect any one outside the package.